waku/v2/peermanager/peer_manager.go

```diff
@@ -73,40 +94,112 @@ func (pm *PeerManager) connectivityLoop(ctx context.Context) {
-			pm.logger.Info("Successfully disconnected connection towards peer", zap.String("peerID", p.String()))
+	outRelayPeers, _ = utils.FilterPeersByProto(pm.host, outPeers, WakuRelayIDv200)
+	pm.logger.Info("Number of Relay peers connected", zap.Int("inRelayPeers", inRelayPeers.Len()),
+		zap.Int("outRelayPeers", outRelayPeers.Len()))
+	return
+}
+
+func (pm *PeerManager) connectToRelayPeers() {
+
+	//Check for out peer connections and connect to more peers.
+	inRelayPeers, outRelayPeers := pm.getRelayPeers()
+	if inRelayPeers.Len() > 0 &&
+		inRelayPeers.Len() > pm.InRelayPeersTarget {
+		pm.pruneInRelayConns(inRelayPeers, outRelayPeers)
+	}
+
+	if outRelayPeers.Len() > pm.OutRelayPeersTarget {
+		return
+	}
+	totalRelayPeers := inRelayPeers.Len() + outRelayPeers.Len()
+	// Establish additional connections if there are peers.
+	//What if the not connected peers in peerstore are not relay peers???
+	if totalRelayPeers < pm.host.Peerstore().Peers().Len() {
+		//Find not connected peers.
+		notConnectedPeers := pm.getNotConnectedPers()
+		//Figure out outside backoff peers.
+
+		//Connect to eligible peers.
+		numPeersToConnect := pm.maxRelayPeers - totalRelayPeers
+
+		//numPeersToConnect = min(min(pm.maxConnections - totalRelayPeers, outsideBackoffPeers.len), MaxParalelDials)
+
+		if numPeersToConnect > notConnectedPeers.Len() {
+			numPeersToConnect = notConnectedPeers.Len() - 1
+		}
+
+		pm.connectToPeers(notConnectedPeers[0:numPeersToConnect])
+	} //Else: Should we raise some sort of unhealthy event??
+}
+
+func (pm *PeerManager) connectToPeers(peers peer.IDSlice) {
+	for _, peerID := range peers {
+		peerInfo := peer.AddrInfo{
+			ID:    peerID,
+			Addrs: pm.host.Peerstore().Addrs(peerID),
+		}
+		pm.peerConnector.publishWork(pm.ctx, peerInfo)
+	}
+}
+
+func (pm *PeerManager) getNotConnectedPers() (notConnectedPeers peer.IDSlice) {
+	for _, peerID := range pm.host.Peerstore().Peers() {
+		if pm.host.Network().Connectedness(peerID) != network.Connected {
+			notConnectedPeers = append(notConnectedPeers, peerID)
+		}
+	}
+	return
+}
+
+func (pm *PeerManager) pruneInRelayConns(inRelayPeers peer.IDSlice, outRelayPeers peer.IDSlice) {
+
+	//Start disconnecting peers, based on what?
+	//For now, just disconnect most recently connected peers
+	//TODO: Need to have more intelligent way of doing this, maybe peer scores.
+	pm.logger.Info("Number of in peer connections exceed targer relay peers, hence pruning",
+		zap.Int("inRelayPeers", inRelayPeers.Len()), zap.Int("inRelayPeersTarget", pm.InRelayPeersTarget))
+	for pruningStartIndex := pm.InRelayPeersTarget; pruningStartIndex < inRelayPeers.Len(); pruningStartIndex++ {
+		p := inRelayPeers[pruningStartIndex]
+		err := pm.host.Network().ClosePeer(p)
+		if err != nil {
+			pm.logger.Warn("Failed to disconnect connection towards peer",
+				zap.String("peerID", p.String()))
 		}
+		pm.logger.Info("Successfully disconnected connection towards peer",
```

Comment:
It probably makes sense to either remove this log line, or change it into `Debug`, since this is what we expect should happen most of the time, it will probably introduce too much noise in the logs
```suggestion
		pm.logger.Info("Successfully disconnected connection towards peer",
```